### PR TITLE
Reine UI

### PR DESCRIPTION
**Pull request by Reine0017**
This is the frontend stuff for all the pages.

### CODE REVIEW

**Review by Reine0017 — COMMENTED** [review]
image


Cinema seats are dropped down (anim). White parts will be changed - might need to add NavBar on top and more info button at bottom right.

**Review by Reine0017 — COMMENTED** [review]
When popup button is pressed (for now), would be changed to on each marker press:
image

On hover at each container:
image

**Review by Reine0017 — COMMENTED** [review]
merged from master and resolved merge conflicts

File: src/main/webapp/aboutus.html [hunk 1]
```diff
@@ -14,6 +14,10 @@
 limitations under the License.
 -->
 
+
```
Comment (jiangyue12392) [inline src/main/webapp/aboutus.html]: Remove extra lines in front and behind

File: src/main/webapp/css/maps-page.css [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  position:fixed;
+  background-color:#313131;
+  overflow:auto
+}
+
+#myMapsHeader{
+  padding-top:50px;
+}
+
+/*
```
Comment (jiangyue12392) [inline src/main/webapp/css/maps-page.css]: Is this still an undecided decision?

File: src/main/webapp/css/selectionMenu.css [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  align-items: center;
+  border: 1px solid #000;
+  position: absolute;
+  top: -2px;
+  left: -2px;
+  right: -2px;
+  bottom: -2px;
+  background-color:brown
+}
+
```
Comment (jiangyue12392) [inline src/main/webapp/css/selectionMenu.css:62]: Extra blank line

File: src/main/webapp/js/maps-page-loader.js [hunk 4]
```diff
@@ -54,83 +62,103 @@ async function createCinemaMarkers(map) {
   });
 }
 
+/** Creates a marker that shows an info window with existing reviews and editable comment session
```
Comment (jiangyue12392) [inline src/main/webapp/js/maps-page-loader.js]: Duplicated comments

File: src/main/webapp/js/maps-page-loader.js [hunk 5]
```diff
@@ -54,83 +62,103 @@ async function createCinemaMarkers(map) {
...
-    map: map
+    map: map,
+    // set the icon as catIcon declared above
+    icon: chairIcon,
+    // must use optimized false for CSS
+    optimized: false
   });
 
-  const infoText = document.createTextNode(cinemaName);
+  /*const infoText = document.createTextNode(cinemaName);
```
Comment (jiangyue12392) [inline src/main/webapp/js/maps-page-loader.js]: Remove dead codes

File: src/main/webapp/js/maps-page-loader.js [hunk 6]
```diff
@@ -54,83 +62,103 @@ async function createCinemaMarkers(map) {
...
 
-  await getMessagesForKey(key, containerDiv);
+  await getMessagesForKey(key);
 
-  const infoWindow = new google.maps.InfoWindow({
-    content: buildInfoWindowInput(map, lat, lng, cinemaName, key, cinemaMarker, containerDiv)
-  });
-  if (updateMode) {
+  buildPopupWindowInput(map, lat, lng, cinemaName, key, cinemaMarker);
+  /*const infoWindow = new google.maps.InfoWindow({
```
Comment (jiangyue12392) [inline src/main/webapp/js/maps-page-loader.js]: Remove dead codes

File: src/main/webapp/js/maps-page-loader.js [hunk 7]
```diff
@@ -54,83 +62,103 @@ async function createCinemaMarkers(map) {
...
-  });
 }
 
 /** Sends a message to the backend for saving. */
 function postMessage(parentKey, text) {
   const params = new URLSearchParams();
   params.append('parentKey', parentKey);
   params.append('text', text);
-  fetch('/marker-messages', {
+    fetch('/marker-messages', {
```
Comment (jiangyue12392) [inline src/main/webapp/js/maps-page-loader.js:112]: Why the indentation change

File: src/main/webapp/js/maps-page-loader.js [hunk 8]
```diff
@@ -1,18 +1,26 @@
...
  * New modes/map styles can be added in extraMapStyles.json file.
  *
  */
-
 function fetchConfigAndBuildMap() {
   const url = '/json/extraMapStyles.json';
   fetch(url)
-    .then(response => response.json())
-    .then(data => addDiffStylesToMap(data.Night));
+      .then(response => response.json())
```
Comment (jiangyue12392) [inline src/main/webapp/js/maps-page-loader.js]: Why the indentation change

File: src/main/webapp/js/maps-page-loader.js [hunk 9]
```diff
@@ -54,83 +62,103 @@ async function createCinemaMarkers(map) {
...
+    fetch('/marker-messages', {
     method: 'POST',
     body: params
   });
 }
 
-async function handleSumbitButtonClick(map, lat, lng, cinemaName, key, marker, text) {
+async function handleSubmitButtonClick(map, lat, lng, cinemaName, key, marker, text) {
   const delay = ms => new Promise(res => setTimeout(res, ms));
+  console.log(key);
```
Comment (jiangyue12392) [inline src/main/webapp/js/maps-page-loader.js]: Remove logging

File: src/main/webapp/js/maps-page-loader.js [hunk 10]
```diff
@@ -54,83 +62,103 @@ async function createCinemaMarkers(map) {
...
+  console.log(text);
   await postMessage(key, text);
-  marker.setMap(null);
+
+  /*marker.setMap(null);
```
Comment (jiangyue12392) [inline src/main/webapp/js/maps-page-loader.js]: Remove dead code

File: src/main/webapp/js/maps-page-loader.js [hunk 11]
```diff
@@ -54,83 +62,103 @@ async function createCinemaMarkers(map) {
...
-  button.appendChild(document.createTextNode('Submit'));
+function buildPopupWindowInput(map, lat, lng, cinemaName, key, marker) {
+  /*const textBox = document.createElement('textarea');*/
+  const textBox = document.getElementById('submitReviewTextArea');
+  /*const button = document.createElement('button');
+  button.appendChild(document.createTextNode('Submit'));*/
+  const button = document.getElementById('submitReviewsButton');
   button.onclick = () => {
-    handleSumbitButtonClick(map, lat, lng, cinemaName, key, marker, textBox.value);
+    console.log(lat,lng,cinemaName,key,marker);
```
Comment (jiangyue12392) [inline src/main/webapp/js/maps-page-loader.js]: Remove logging

File: src/main/webapp/js/maps-page-loader.js [hunk 12]
```diff
@@ -54,83 +62,103 @@ async function createCinemaMarkers(map) {
...
+  createCinemaMarkerForDisplay(map, lat, lng, cinemaName, key, updateMode=true);*/
 }
 
 /** Builds and returns HTML elements that show an editable textbox and a submit button. */
-function buildInfoWindowInput(map, lat, lng, cinemaName, key, marker, containerDiv) {
-  const textBox = document.createElement('textarea');
-  const button = document.createElement('button');
-  button.appendChild(document.createTextNode('Submit'));
+function buildPopupWindowInput(map, lat, lng, cinemaName, key, marker) {
+  /*const textBox = document.createElement('textarea');*/
```
Comment (jiangyue12392) [inline src/main/webapp/js/maps-page-loader.js]: Remove dead code

File: src/main/webapp/js/maps-page-loader.js [hunk 13]
```diff
@@ -54,83 +62,103 @@ async function createCinemaMarkers(map) {
...
+    handleSubmitButtonClick(map, lat, lng, cinemaName, key, marker, textBox.value);
   };
-  containerDiv.appendChild(document.createElement('br'));
+  /*containerDiv.appendChild(document.createElement('br'));
   containerDiv.appendChild(textBox);
   containerDiv.appendChild(document.createElement('br'));
   containerDiv.appendChild(document.createElement('br'));
-  containerDiv.appendChild(button);
-  return containerDiv;
+  containerDiv.appendChild(button);*/
```
Comment (jiangyue12392) [inline src/main/webapp/js/maps-page-loader.js]: Remove dead code

File: src/main/webapp/js/mapsPopUp.js [hunk 14]
```diff
@@ -0,0 +1,17 @@
+/*// Validating Empty Field
```
Comment (jiangyue12392) [inline src/main/webapp/js/mapsPopUp.js]: Dead code or u want to implement it?

File: src/main/webapp/user-page.html [hunk 15]
```diff
@@ -21,32 +21,41 @@
...
+          <li><a href="/aboutus.html">About Our Team</a></li>
+          <li><a href="/feed.html">Public Feed Page</a></li>
+          <li><a href="/maps.html">Maps Page</a></li>
+        </ul>
+      </nav>
+    </header>
     <h1 id="page-title">User Page</h1>
 
-    <form id="message-form" enctype="multipart/form-data" method="POST" class="hidden">
+    <!--<form id="message-form" enctype="multipart/form-data" method="POST" class="hidden">
```
Comment (jiangyue12392) [inline src/main/webapp/user-page.html]: Dead code

File: src/main/webapp/user-page.html [hunk 16]
```diff
@@ -21,32 +21,41 @@
...
       Upload an image:
       <input type="file" name="image">
-      <br />
+      <br/>
       <input type="submit" value="Submit">
     </form>
     <hr/>
 
-    <div id="message-container">Loading...</div>
+    <div id="message-container">Loading...</div>-->
```
Comment (jiangyue12392) [inline src/main/webapp/user-page.html]: Dead code

**Review by jiangyue12392 — APPROVED** [review]
Looks good to me!

**Pull request merged**